/*
 * Created by helen chan  on 06/11/2021
 * Email: kaibaking@gmail.com
 * 
 */
syntax = "proto3";

//import "google/protobuf/empty.proto";

package ECSSE;

service RPC {

// Setup
rpc setup (SetupRequestMessage) returns ( ExecuteStatus ) {}

// Search
rpc search (SearchRequestMessage) returns (stream SearchReply) {}

// Update
rpc update (UpdateRequestMessage) returns ( ExecuteStatus ) {}
rpc batch_update (stream UpdateRequestMessage) returns ( ExecuteStatus ) {}

// only used for expriment simulation
rpc cache (CacheRequestMessage) returns ( ExecuteStatus ) {}
rpc batch_cache (stream CacheRequestMessage) returns ( ExecuteStatus ) {}

}

message SetupRequestMessage
{
    uint64 setup_size = 1;
}

message SearchRequestMessage
{
    bytes st = 1;
	uint64 uc = 2; 
}


message SearchReply
{
	bytes ind = 1;
}

message CacheRequestMessage
{
    bytes inds = 1;
}

message UpdateRequestMessage
{
    bytes l = 1;
    bytes e = 2;
}

message ExecuteStatus
{
	bool status = 1; 
}
